### PR TITLE
Update Security for CHIPS (Partitioned Cookies)

### PR DESCRIPTION
**Pull request by barbj**
This is for SSO and core security updates for CHIPS Partitioned cookie attribute.

### CODE REVIEW

**Comment by barbj** [comment]
#build

**Comment by barbj** [comment]
#build 
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.oidc.client_fat.4,com.ibm.ws.security.oidc.client_fat.backchannelLogout.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.2,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.3,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.1,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.2,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.commonTest.LibertyOP

**Comment by barbj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.oidc.client_fat.4,com.ibm.ws.security.oidc.client_fat.backchannelLogout.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.2,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.3,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.1,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.2,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.commonTest.LibertyOP

**Comment by barbj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.oidc.client_fat.4,com.ibm.ws.security.oidc.client_fat.backchannelLogout.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.2,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.3,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.1,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.2,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.commonTest.LibertyOP

**Comment by barbj** [comment]
I added Partitioned tests where there are SameSite tests.  I added Partitioned tolerance where there is SameSite tolerance.  I also fixed up some of the SameSite tolerance to alternate values.

**Comment by barbj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.oidc.client_fat.4,com.ibm.ws.security.oidc.client_fat.backchannelLogout.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.2,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.3,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.1,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.2,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.commonTest.LibertyOP

**Comment by barbj** [comment]
Hi @ayoho , I'm ready for review again.  Can you take a look?  New PR, same code :) 

I'm ignoring the failure in `testRescheduleUnderConfigUpdateRun_Lite` because it only happened 1 time out of the 5 builds that I did, and 3 of them only modified a `metadata.properties` file.

**Comment by barbj** [comment]
Linking to #27405

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/ReferrerURLCookieHandler.java [hunk 1]
```diff
@@ -113,8 +110,22 @@ public Cookie createCookie(String cookieName, @Sensitive String value, boolean e
...
             if (sameSite.equals("None")) {
                 c.setSecure(true);
+				Boolean partitioned = webAppSecConfig.getPartitionedCookie();
+                if (partitioned!=null) {
+				    //web container wants the value as a n/v pair
+                    requestState.setCookieAttributes(cookieName, "Partitioned="+partitioned.toString());
+                }
             }
-        }
+		} else if (sameSite == null) {
```
Comment (volosied) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/ReferrerURLCookieHandler.java]: sameSite == null || sameSite.equals("Disabled")

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java [hunk 2]
```diff
@@ -174,8 +171,22 @@ protected Cookie createCookie(HttpServletRequest req, String cookieName, String
...
             if (sameSite.equals("None")) {
                 ssoCookie.setSecure(true);
+			 	Boolean partitioned = config.getPartitionedCookie();
+                if (partitioned!=null) {
+				    //web container wants the value as a n/v pair
+                    requestState.setCookieAttributes(cookieName, "Partitioned="+partitioned.toString());
+                }
             }
-        }
+        } else if (sameSite == null) {
```
Comment (volosied) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java]: sameSite == null || sameSite.equals("Disabled")

File: dev/com.ibm.ws.security.common/src/com/ibm/ws/security/common/web/JavaScriptUtils.java [hunk 3]
```diff
@@ -227,6 +224,9 @@ public Map<String, String> getWebAppSecurityConfigCookieProperties(WebAppSecurit
                 cookieProperties.put("SameSite", sameSite);
                 if ("None".equalsIgnoreCase(sameSite)) {
                     cookieProperties.put("secure", null);
+			    	if (webAppSecurityConfig.isPartitionedCookie()) {
+              			cookieProperties.put("Partitioned", null);
+            		}
```
Comment (ayoho) [inline dev/com.ibm.ws.security.common/src/com/ibm/ws/security/common/web/JavaScriptUtils.java]: Remember to come back and swap out tabs for spaces.

**Comment by barbj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.oidc.client_fat.4,com.ibm.ws.security.oidc.client_fat.backchannelLogout.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.2,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.3,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.1,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.2,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.commonTest.LibertyOP

**Comment by barbj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.oidc.client_fat.4,com.ibm.ws.security.oidc.client_fat.backchannelLogout.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.2,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.3,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.1,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.2,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.commonTest.LibertyOP

**Comment by barbj** [comment]
PB that includes schema validator updates in CD-OPEN:
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_3QoKcBj9Ee-aC7a28EbkCw

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/ReferrerURLCookieHandler.java (hunk 1 above)
Comment (barbj) [reply dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/ReferrerURLCookieHandler.java]: Doesn't that make it just a regular **else**?

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java (hunk 2 above)
Comment (barbj) [reply dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java]: Doesn't that make it just a regular else?

**Comment by barbj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.oidc.client_fat.4,com.ibm.ws.security.oidc.client_fat.backchannelLogout.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.2,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.3,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.1,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.2,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.commonTest.LibertyOP

**Comment by barbj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.1,com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.oidc.client_fat.4,com.ibm.ws.security.oidc.client_fat.backchannelLogout.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.2,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.1,com.ibm.ws.security.oidc.client_fat.backchannelLogout.saml.2,com.ibm.ws.security.oidc.client_fat.claimPropagation,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.backchannelLogout,com.ibm.ws.security.oidc.server_fat.jaxrs.config.commonTest,com.ibm.ws.security.oidc.server_fat.jaxrs.config.noOP,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oauth,com.ibm.ws.security.oidc.server_fat.jaxrs.config.oidc,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.spnego,com.ibm.ws.security.social_fat.LibertyOP.1,com.ibm.ws.security.social_fat.LibertyOP.2,com.ibm.ws.security.social_fat.LibertyOP.3,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.1,com.ibm.ws.security.social_fat.LibertyOP.backchannelLogout.2,com.ibm.ws.security.social_fat.LibertyOP.claimPropagation,com.ibm.ws.security.social_fat.commonTest.LibertyOP

**Comment by barbj** [comment]
Hi @ayoho , this PR is ready to be reviewed again for reals!

File: dev/com.ibm.ws.webcontainer.security.admin/src/com/ibm/ws/webcontainer/security/admin/internal/WebAdminSecurityConfigImpl.java [hunk 4]
```diff
@@ -301,17 +302,42 @@ public String getBasicAuthRealmName() {
...
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isPartitionedCookie() {
+        WebAppSecurityConfig globalConfig = WebAppSecurityCollaboratorImpl.getGlobalWebAppSecurityConfig();
+        if (globalConfig != null)
+            return globalConfig.isPartitionedCookie();
+       
+        if (partitionedCookie!=null && partitionedCookie==Boolean.TRUE) {
```
Comment (ayoho) [inline dev/com.ibm.ws.webcontainer.security.admin/src/com/ibm/ws/webcontainer/security/admin/internal/WebAdminSecurityConfigImpl.java]: Does `partitionedCookie` ever get updated? Will it ever not be null?

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 5]
```diff
@@ -118,3 +118,11 @@ loggedOutCookieCacheCacheRef.desc=The JCache cache reference that is used as the
...
 useContextRootForSSOCookiePath=Use the context root for the SSO cookie path
 useContextRootForSSOCookiePath.desc=Specifies that the cookie path equals the context root of the web module instead of the backslash character (/).
+
+
+ # The words Partitioned, SameSite, and None should not be translated
+partitionedCookie=Partition SSO cookies with SameSite set to none
+partitionedCookie.desc=Specifies whether to partition SSO cookies with SameSite set to None and therefore send them in cross-site requests. By default, the HTTP transport settings determine whether to partition SSO cookies.
+partitionedCookieTrue=Always partition SSO cookies that have SameSite set to None.
+partitionedCookieFalse=Do not partition SSO cookies.
+partitionedCookieDefer=Allow the HTTP transport settings to determine whether to partitioned SSO cookies.
```
Comment (ayoho) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]:
```suggestion
partitionedCookieDefer=Allow the HTTP transport settings to determine whether to partition SSO cookies.
```

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/metatype/metatype.xml [hunk 6]
```diff
@@ -139,6 +139,13 @@
      			<Option label="%sameSiteCookieStrict" value="Strict"/>
         </AD>    
         
+        <AD id="partitionedCookie" name="%partitionedCookie" description="%partitionedCookie.desc"
+            required="false" type="String" default="Defer">
+            <Option label="%partitionedCookieTrue"  value="True"/>
+            <Option label="%partitionedCookieFalse" value="False"/>
+            <Option label="%partitionedCookieDefer" value="Defer"/>
```
Comment (ayoho) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/metatype/metatype.xml]: Might be a nit, but I'd prefer to have the values all be lowercase: `"true"`, `"false"`, `"defer"`.

File: dev/com.ibm.ws.webcontainer.security.feature/src/com/ibm/ws/webcontainer/security/feature/internal/FeatureWebSecurityConfigImpl.java [hunk 7]
```diff
@@ -406,11 +407,34 @@ public String getBasicAuthRealmName() {
...
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isPartitionedCookie() {
+        WebAppSecurityConfig globalConfig = WebAppSecurityCollaboratorImpl.getGlobalWebAppSecurityConfig();
+        if (globalConfig != null)
+            return globalConfig.isPartitionedCookie();
+       
+        if (partitionedCookie!=null && partitionedCookie==Boolean.TRUE) {
```
Comment (ayoho) [inline dev/com.ibm.ws.webcontainer.security.feature/src/com/ibm/ws/webcontainer/security/feature/internal/FeatureWebSecurityConfigImpl.java]: Same as before - will `partitionedCookie` ever not be null?

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/ReferrerURLCookieHandler.java [hunk 8]
```diff
@@ -113,8 +110,26 @@ public Cookie createCookie(String cookieName, @Sensitive String value, boolean e
 
             if (sameSite.equals("None")) {
                 c.setSecure(true);
+				Boolean partitioned = webAppSecConfig.getPartitionedCookie();
```
Comment (ayoho) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/ReferrerURLCookieHandler.java:113]: Mixture of tabs and spaces here.

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java [hunk 9]
```diff
@@ -174,8 +171,26 @@ protected Cookie createCookie(HttpServletRequest req, String cookieName, String
 
             if (sameSite.equals("None")) {
                 ssoCookie.setSecure(true);
+			 	Boolean partitioned = config.getPartitionedCookie();
```
Comment (ayoho) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java]: Mixed tabs and spaces.

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/internal/WebAppSecurityConfigImpl.java [hunk 10]
```diff
@@ -603,4 +630,19 @@ public long postParamMaxRequestBodySize() {
...
+    // This method is for config users that need to know if an admin has provided a true/false or no value for a
+    // boolean config attribute.  The current infrastructure does not properly support this
+    public static Boolean getBooleanValue(String attribute, String strValue) {
+      Boolean retVal = null;
+      if (strValue!=null && strValue.length()>0) {
+        if ("true".equalsIgnoreCase(strValue) || "false".equalsIgnoreCase(strValue)) {
+          retVal = Boolean.valueOf(strValue);
+        } else {
+          Object[] inserts = new Object[] { strValue, (attribute==null?"":attribute)}; 
+          Tr.warning(tc, "INVALID_BOOLEAN_ATTRIBUTE", inserts);
```
Comment (ayoho) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/internal/WebAppSecurityConfigImpl.java]: Where does the `INVALID_BOOLEAN_ATTRIBUTE` NLS message exist?

**Comment by barbj** [comment]
#build

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/internal/WebAppSecurityConfigImpl.java (hunk 10 above)
Comment (barbj) [reply dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/internal/WebAppSecurityConfigImpl.java]: I removed that else section.

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java (hunk 9 above)
Comment (barbj) [reply dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/SSOCookieHelperImpl.java]: fixed

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/ReferrerURLCookieHandler.java (hunk 8 above)
Comment (barbj) [reply dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/ReferrerURLCookieHandler.java:113]: fixed

File: dev/com.ibm.ws.webcontainer.security.feature/src/com/ibm/ws/webcontainer/security/feature/internal/FeatureWebSecurityConfigImpl.java (hunk 7 above)
Comment (barbj) [reply dev/com.ibm.ws.webcontainer.security.feature/src/com/ibm/ws/webcontainer/security/feature/internal/FeatureWebSecurityConfigImpl.java]: removed this condition

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/metatype/metatype.xml (hunk 6 above)
Comment (barbj) [reply dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/metatype/metatype.xml]: changed to all LC

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties (hunk 5 above)
Comment (barbj) [reply dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: changed partitioned to partition

File: dev/com.ibm.ws.webcontainer.security.admin/src/com/ibm/ws/webcontainer/security/admin/internal/WebAdminSecurityConfigImpl.java (hunk 4 above)
Comment (barbj) [reply dev/com.ibm.ws.webcontainer.security.admin/src/com/ibm/ws/webcontainer/security/admin/internal/WebAdminSecurityConfigImpl.java]: removed condition

File: dev/com.ibm.ws.security.common/src/com/ibm/ws/security/common/web/JavaScriptUtils.java (hunk 3 above)
Comment (barbj) [reply dev/com.ibm.ws.security.common/src/com/ibm/ws/security/common/web/JavaScriptUtils.java]: fixed

**Review by ayoho — DISMISSED** [review]

**Comment by barbj** [comment]
#build

**Review by ayoho — APPROVED** [review]

**Comment by barbj** [comment]
Hi @volosied , this should be ready to be reviewed for the last time.

**Comment by barbj** [comment]
#libby : check headers

**Pull request merged**